README.md

```diff
@@ -1,7 +1,295 @@
+
+- [Overview](#overview)
+  - [Algebraic effects](#algebraic-effects)
+  - [Higher-order effects](#higher-order-effects)
+  - [Fusion](#fusion)
+- [Usage](#usage)
+  - [Using built-in effects](#using-built-in-effects)
+  - [Running effects](#running-effects)
+  - [Defining new effects](#defining-new-effects)
+  - [Defining effect handlers](#defining-effect-handlers)
+- [Benchmarks](#benchmarks)
+- [Related work](#related-work)
+  - [Comparison to `mtl`](#comparison-to--mtl-)
+  - [Comparison to `freer-simple`](#comparison-to--freer-simple-)
+
+
+## Overview
+
+`higher-order-effects` is an effect system for Haskell emphasizing expressivity and efficiency. The former is achieved by encoding [algebraic](#algebraic-effects), [higher-order](#higher-order-effects) effects, while the latter is the result of [fusing](#fusion) effect handlers all the way through computations.
+
+Readers already familiar with effect systems may wish to start with the [usage](#usage) instead.
+
+
+### Algebraic effects
+
+In `higher-order-effects` and other systems with _algebraic_ (or, sometimes, _extensible_) effects, effectful programs are split into two parts: the specification (or _syntax_) of the actions to be performed, and the interpretation (or _semantics_) given to them. Thus, a program written using the syntax of an effect can be given different meanings by using different effect handlers.
+
+These roles are performed by the effect and carrier types, respectively. Effects are datatypes with one constructor for each action. Carriers are generally `newtype`s, with a `Carrier` instance specifying how an effect’s constructors should be interpreted. Each carrier handles one effect, but multiple carriers can be defined for the same effect, corresponding to different interpreters for the effect’s syntax.
+
+
+### Higher-order effects
+
+Unlike most other effect systems, `higher-order-effects` offers _higher-order_ (or _scoped_) effects in addition to first-order algebraic effects. In a strictly first-order algebraic effect system, operations (like `local` or `catchError`) which specify some action limited to a given scope must be implemented as interpreters, hard-coding their meaning in precisely the manner algebraic effects were designed to avoid. By specifying effects as higher-order functors, these operations are likewise able to be given a variety of interpretations. This means, for example, that you can introspect and redefine both the `local` and `ask` operations provided by the `Reader` effect, rather than solely `ask` (as is the case with certain formulations of algebraic effects).
+
+As Nicolas Wu et al showed in _[Effect Handlers in Scope][]_, this has implications for the expressiveness of effect systems. It also has the benefit of making effect handling more consistent, since scoped operations are just syntax which can be interpreted like any other, and are thus simpler to reason about.
+
+
+### Fusion
```

Comment:
@thumphries [suggests](https://twitter.com/thumphriees/status/1056644417199591424) spending some words here to note that it’s not just `RULES` but a bit more durable than that.